webapp/src/types/mattermost-webapp/index.d.ts

```diff
@@ -1,5 +1,35 @@
+import React from 'react';
+
+export type DateTimeConfig = {
+    date: string,
+    time: string,
+}
+
+export type CustomConfigTypes = DateTimeConfig;
+
+export type Config = {
+    PluginSettings?: {
+        Plugins?: {
+            'com.mattermost.user-survey'?: {
+                surveydatetime: DateTimeConfig,
+            },
+        },
+    },
+    [key: string]: unknown,
```

Comment:
Can we either import this type from `mattermost-redux` or get rid of this arbitrary dictionary? I always find they just kinda muddle/defeat the purpose of typing, unless we explicitly expect to find any potential value in there.

Reply:
It could extend `GlobalState` from `@mattermost/types`. That only contains types for `state.entities`, but that's most of what you'd need anyway

Reply:
Checking the value received in here, its `AdminConfig`, so set it to that.